daemon/start.go

```diff
@@ -117,10 +116,6 @@ func (daemon *Daemon) containerStart(container *container.Container) (err error)
 		return err
 	}
 
-	// Make sure NetworkMode has an acceptable value. We do this to ensure
-	// backwards API compatibility.
-	container.HostConfig = runconfig.SetDefaultNetModeIfBlank(container.HostConfig)
-
```

Comment:
Maybe this is still need? if a container was created and never be started by old  docker, and then start by new docker with your changes, would that be a problem?